mypy/errors.py

```diff
@@ -310,7 +310,11 @@ def report(self,
         self.add_error_info(info)
 
     def _add_error_info(self, file: str, info: ErrorInfo) -> None:
-        assert file not in self.flushed_files
+        if file in self.flushed_files:
+            self.report(info.line, None,
+                        "Trying to report an error in a file that has already been flushed."
+                        " Skipping...", severity='warning', file=file, only_once=True)
+            return
```

Comment:
Hmm... I think I want to return here so that the errors don't get logged multiple times. But that means this warning gets skipped too :/